internal/controller/nonadminbackup_controller.go

```diff
@@ -255,65 +255,86 @@ func (r *NonAdminBackupReconciler) CreateVeleroBackupSpec(ctx context.Context, l
 		veleroBackup = velerov1api.Backup{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      veleroBackupName,
-				Namespace: constant.OadpNamespace,
+				Namespace: r.OADPNamespace,
 			},
 			Spec: *backupSpec,
 		}
-	} else if err != nil {
-		logger.Error(err, "Unable to fetch VeleroBackup")
-		return true, false, err
-	} else {
-		// We should not update already created VeleroBackup object.
-		// The VeleroBackup within NonAdminBackup will
-		// be reverted back to the previous state - the state which created VeleroBackup
-		// in a first place, so they will be in sync.
-		logger.Info("Backup already exists, updating NonAdminBackup status", nameField, veleroBackupName)
-		updatedNab, errBackupUpdate := function.UpdateNonAdminBackupFromVeleroBackup(ctx, r.Client, logger, nab, &veleroBackup)
-		// Regardless if the status was updated or not, we should not
-		// requeue here as it was only status update.
-		if errBackupUpdate != nil {
-			return true, false, errBackupUpdate
-		} else if updatedNab {
-			logger.V(1).Info("NonAdminBackup CR - Rqueue after Status Update")
-			return false, true, nil
+
+		// Ensure labels are set for the Backup object
+		existingLabels := veleroBackup.Labels
+		naManagedLabels := function.AddNonAdminLabels(existingLabels)
+		veleroBackup.Labels = naManagedLabels
+
+		// Ensure annotations are set for the Backup object
+		existingAnnotations := veleroBackup.Annotations
+		ownerUUID := string(nab.ObjectMeta.UID)
+		nabManagedAnnotations := function.AddNonAdminBackupAnnotations(nab.Namespace, nab.Name, ownerUUID, existingAnnotations)
+		veleroBackup.Annotations = nabManagedAnnotations
+
+		err = r.Create(ctx, &veleroBackup)
```

Comment:
I am thinking we should use controller-runtime [`CreateOrPatch` ](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#CreateOrPatch)instead of `Create` call here, to avoid race conditions and resource overwrites.

Reply:
This is the case where we checked explicitly if the Velero Backup exists or not.

In case there is possible race condition I think the correct way is to use [RetryOnConflict](https://pkg.go.dev/k8s.io/client-go/util/retry#RetryOnConflict)

Reply:
Create or Patch does the same thing. Reference https://github.com/kubernetes-sigs/controller-runtime/blob/v0.19.0/pkg/controller/controllerutil/controllerutil.go#L346

Here I am not worried by race conditions. Examples:
- If it fails for unknown reason, it will reconcile again and try again
- It if fails because something else created it, it will not try to create it again, and will go to next step (update phase) in next reconcile